torchinductor/codegen/cpp.py

```diff
@@ -497,34 +490,51 @@ def __init__(self, scheduler):
     def group_fn(self, sizes):
         return tuple(tuple(map(V.graph.sizevars.simplify, s)) for s in sizes)
 
-    def codegen(self, *groups):
-        group, reduction_group = groups
-
+    @staticmethod
+    def can_fuse_horizontal(node1, node2):
+        _, (vars1, reduce1) = node1.group
+        _, (vars2, reduce2) = node2.group
+        if vars1 == vars2 and reduce1 == reduce2:
+            return True
+        if reduce1 == () and vars1 == vars2 + reduce2:
```

Comment:
Should this be a + or a *?

Reply:
In the C++ backend the group is a tuple of all the iteration ranges.  We don't flatten to numel.  For example:

C++ group: `((a,b,c), ())`

Triton group: `(a*b*c, 1)`